posts/2018-02-09-serverless-and-graphql-in-love.md

```diff
@@ -0,0 +1,611 @@
+
+```yml
+import { OAuth2 } from 'oauth';
+const Twitter = require('twitter');
+
+async function getFollowing(handle, consumerKey, consumerSecret) {
+  const url = 'friends/list';
+
+  const oauth2 = new OAuth2(
+    consumerKey,
+    consumerSecret,
+    'https://api.twitter.com/',
+    null,
+    'oauth2/token',
+    null
+  );
+
+  return new Promise(resolve => {
+    oauth2.getOAuthAccessToken(
+      '',
+      {
+        grant_type: 'client_credentials',
+      },
+      (error, accessToken) => {
+        resolve(accessToken);
+      }
+    );
+  })
+    .then(accessToken => {
+      const client = new Twitter({
+        consumer_key: consumerKey,
+        consumer_secret: consumerSecret,
+        bearer_token: accessToken,
+      });
+
+      const params = { screen_name: handle };
+
+      return client
+        .get(url, params)
+      //then parse the result
+}
+```
+
+**Note:** A complete example is given [here](https://raw.githubusercontent.com/serverless/serverless-graphql/master/app-backend/rest-api/resolvers.js). You can also look at Saeri's blog post on building a [Serverless GraphQL Gateway on top of a 3rd Party REST API](https://serverless.com/blog/3rd-party-rest-api-to-graphql-serverless/).
+
+Go ahead and run it locally on your machine:
+
+```
+cd app-backend/rest-api
+yarn install
+yarn start
+```
+
+And deploy to production:
+
+```
+cd app-backend/rest-api
+yarn deploy-prod
+``` 
+ 
+## Client Integrations (Apollo ReactJS, Netlify, and S3)
+
+This repository comes with two client implementations
+
+1. [Apollo Client 2.0](https://dev-blog.apollodata.com/apollo-client-2-0-beyond-graphql-apis-888807b53afe)
+2. [AppSync Client](https://dev-blog.apollodata.com/aws-appsync-powered-by-apollo-df61eb706183)
+
+If you are new to the ReactJs + Apollo Integration, I would recommend going through [these tutorials](https://www.learnapollo.com/tutorial-react/react-01/).
+
+The code for apollo-client in the serverless-graphql repo is [here](https://github.com/serverless/serverless-graphql/tree/master/app-client/apollo-client).
+
+To start the client on local, first, start any backend service on local. For example:
+
+```
+cd app-backend/rest-api
+yarn install
+yarn start
+```
+
+Now, make sure `http://localhost:4000/graphiql` is working.
+
+Now, kickstart Apollo Client (as shown below), you will have a react server running on your local machine. The setup is created using [create react app](https://github.com/facebook/create-react-app):
+
+```
+cd app-client/apollo-client
+yarn install
+yarn start
+```
+
+![!Live Example](https://user-images.githubusercontent.com/1587005/36068493-de82620e-0e8b-11e8-887b-e1593cd3c8cc.gif)
+
+In production, you can also deploy the client on Netlify or AWS S3. Please follow the instructions [here](https://github.com/serverless/serverless-graphql#setup-for-production-deploy-resources-to-aws).
+
+## Performance Analysis 
```

Comment:
Would be nice to have some fancy charts here (if possible or even necessary)...

Reply:
I added 3 charts ;)